cranelift/codegen/src/isa/aarch64/lower.rs

```diff
@@ -1415,22 +1435,69 @@ pub(crate) fn lower_icmp<C: LowerCtx<I = Inst>>(
                 }
             }
         }
+        condcode
     } else if !ty.is_vector() {
```

Comment:
Nit - the code structure looks like this:
```
  ...
} else if !cond {
  ...
} else {
  ...
```
Since you are changing the surrounding code significantly, could you also transform this, so that it does not use a negated condition:
```
  ...
} else if cond {
  ...
} else {
  ...
```